### DISCUSSION TITLE
How to properly disable LSP formatting?

### DISCUSSION THREAD

**Discussion by rorj** [LazyVim/LazyVim#2159] [Q&A] [answered]
After [latest changes](https://github.com/neovim/nvim-lspconfig/commit/fdf738ace40ccde021b77f8a1e90a95c7b8c2f1e) to cssls config I can't figure out how to disable auto formatting with that particular server. My first idea was changing the settings
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        cssls = {
          settings = {
            css = {
              format = {
                enable = false,
              },
            },
            scss = {
              format = {
                enable = false,
              },
            },
          },
        },
      },
    },
  },
}
```
unfortunately it doesn't work. What am I missing?

**Comment by dpetka2001** [comment, marked as answer]
```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        cssls = {
          init_options = {
            provideFormatter = false,
          },
        },
      },
    },
  },
```
This should do it I believe. You can check with `LazyFormatInfo` and `ConformInfo` when you're in a `.css` file to show you the available formatters if you want to disable something else.

**Reply by rorj** [reply]
Yep, that's it, thanks! Working (or rather not working) as expected now. `LazyFormatInfo` seems very useful for debugging such issues as well.
